ast/parser.go

```diff
@@ -156,8 +160,10 @@ type ParserOptions struct {
 }
 
 // EffectiveRegoVersion returns the effective RegoVersion to use for parsing.
-// Deprecated: Use RegoVersion instead.
-func (po *ParserOptions) EffectiveRegoVersion() RegoVersion {
+func (po ParserOptions) EffectiveRegoVersion() RegoVersion {
+	if po.RegoVersion == RegoUndefined {
```

Comment:
Should RegoVersion still be a public field?

Reply:
If private, it's match the toStringOpts more closely too.

Reply:
This needs to remain public I'm afraid. There is no factory for `ast.ParserOptions`, so the `RegoVersion` field is how existing code would set the rego-version. Changing this would be a big breaking change.